src/yunohost/data_migrations/0009_setup_group_permission.py

```diff
@@ -0,0 +1,142 @@
+from moulinette.core import init_authenticator
+from yunohost.utils.error import YunohostError
+from moulinette.utils.log import getActionLogger
+
+from yunohost.tools import Migration
+from yunohost.utils.filesystem import free_space_in_directory, space_used_by_directory
+from yunohost.user import user_list, user_group_add, user_group_update
+from yunohost.app import app_setting, app_list
+from yunohost.service import service_regen_conf
+from yunohost.permission import permission_add, permission_sync_to_user
+from yunohost.user import user_permission_add
+
+logger = getActionLogger('yunohost.migration')
+
+###################################################
+# Tools used also for restoration
+###################################################
+
+def migrate_LDAP_db(auth):
+    logger.info(m18n.n("migration_0009_update_LDAP_database"))
+    try:
+        auth.remove('cn=sftpusers,ou=groups')
+    except:
+        logger.warn(m18n.n("error_when_removing_sftpuser_group"))
+
+    with open('/usr/share/yunohost/yunohost-config/moulinette/ldap_scheme.yml') as f:
+        ldap_map = yaml.load(f)
+
+    try:
+        attr_dict = ldap_map['parents']['ou=permission']
+        auth.add('ou=permission', attr_dict)
+
+        attr_dict = ldap_map['children']['cn=all_users,ou=groups']
+        auth.add('cn=all_users,ou=groups', attr_dict)
+
+        for rdn, attr_dict in ldap_map['depends_children'].items():
+            auth.add(rdn, attr_dict)
+    except Exception as e:
+        raise YunohostError("migration_0009_LDAP_update_failed", error=e)
+
+    logger.info(m18n.n("migration_0009_create_group"))
+
+    #Create a group for each yunohost user
+    user_list = auth.search('ou=users,dc=yunohost,dc=org',
+                            '(&(objectclass=person)(!(uid=root))(!(uid=nobody)))',
+                            ['uid', 'uidNumber'])
+    for user_info in user_list:
+        username = user_info['uid'][0]
+        auth.update('uid=%s,ou=users' % username,
+                    {'objectClass': ['mailAccount', 'inetOrgPerson', 'posixAccount', 'userPermissionYnh']})
+        user_group_add(auth, username, gid=user_info['uidNumber'][0], sync_perm=False)
+        user_group_update(auth, groupname=username, add_user=username, force=True, sync_perm=False)
+        user_group_update(auth, 'all_users', add_user=username, force=True, sync_perm=False)
+
+
+def migrate_app_permission(auth, app=None):
+    logger.info(m18n.n("migration_0009_migrate_permission"))
+
+    if app:
+        apps = app_list(installed=True, filter=app)['apps']
+    else:
+        apps = app_list(installed=True)['apps']
+
+    for app_info in apps:
+        app = app_info['id']
+        permission = app_setting(app, 'allowed_users')
+        path = app_setting(app, 'path')
+        domain = app_setting(app, 'domain')
+
+        url = None
+        if domain and path:
+            url = domain + path
+        permission_add(auth, app, 'main', urls=[url], default_allow=True, sync_perm=False)
```

Comment:
`permission="main"` ? :S

Reply:
Yes because the main permission for each apps is "main".

Reply:
I mean my point was to add the `permission=` to be more explicit :s

Reply:
Ok